sources/Queue.cpp

```diff
@@ -0,0 +1,60 @@
+#include <queue.hpp>
+
+void Construct(Queue& queue)
+{
+    queue.Head = nullptr;
+    queue.Tail = nullptr;
+}
+void Destruct(Queue& queue)
+{
+    Queue::Node* temp;
+    while (queue.Head != nullptr)
+    {
+        temp = queue.Head;
+        queue.Head = queue.Head->Next;
+        delete temp;
+    }
+    queue.Head = nullptr;
+    queue.Tail = nullptr;
+}
+
+void Push(Queue& queue, int value)
+{
+    if (queue.Head == nullptr){
+        queue.Head = new Queue::Node();
+        queue.Head->Data = value;
+        queue.Head->Next = nullptr;
+        queue.Tail = queue.Head;
+    }
+    else
+    {
+        queue.Tail->Next = new Queue::Node();
+        queue.Tail = queue.Tail->Next;
+        queue.Tail->Next = nullptr;
+        queue.Tail->Data = value;
+    }
+}
+int Pop(Queue& queue)
+{
+    if (queue.Head != nullptr)
+    {
+        int result = queue.Head->Data;
+        if (queue.Head == queue.Tail){
+            delete queue.Head;
+            queue.Head = nullptr;
+            queue.Tail = nullptr;
+        }
+        else
+        {
+            Queue::Node* temp = queue.Head;
+            queue.Head = queue.Head->Next;
+            delete temp;
+        }
+        return result;
+    }
```

Comment:
Есть такой паттерн программирования: guard clause. Если формализовать - то индентацию (и соответственно цикломатическую и когнитивную сложность кода) можно уменьшить, если проверять крайние случаи, а не их отсутствие. На практике гораздо легче понимать, что условие можно развернуть в другую сторону, и поменять местами ветки хода исполнения. В условии оказывается короткая ветка, а вне - длинная:
```suggestion
    if (queue.Head == nullptr) {
      return 0;
    }
    int result = queue.Head->Data;
    if (queue.Head == queue.Tail){
      delete queue.Head;
      queue.Head = nullptr;
      queue.Tail = nullptr;
    } else {
       Queue::Node* temp = queue.Head;
       queue.Head = queue.Head->Next;
       delete temp;
    }
```
Заметьте, что индентация уменьшилась, а код стал проще в понимании